simulator.py

```diff
@@ -0,0 +1,169 @@
+"""
+Create a simulator to calculate pnl
+"""
+
+import collections
+import copy
+import random
+from thousandaire.constants import TRADING_CONFIG
+from thousandaire.dataset import Data
+
+DATASET = 'price_dataset'
+COST_FUNC = 'cost_function'
+
+class SimulateMethod():
+    """
+    The method for simulating.
+    """
+    def __init__(self, region, instruments):
+        self.quantity = collections.namedtuple('value', ['value', 'quantity'])
+        self.cur_asset = {
+            instrument : self.quantity(0, 0) for instrument in instruments}
+        self.cur_asset[region] = self.quantity(1, 1)
+        self.__portfolio = list()
+        self.__pnl = Data('pnl', ['pnl', 'trading_cost'])
+
+    def get_results(self):
+        """
+        Return results.
+        """
+        result = Data('result', ['pnl', 'trading_cost', 'position'])
```

Comment:
Since we have two versions of positions (`positions_raw`, and `positions_np`) now, can you implement both them in this code as well?